Tests/NIOPosixTests/ChannelPipelineTest.swift

```diff
@@ -2693,6 +2693,132 @@ class ChannelPipelineTest: XCTestCase {
 
         XCTAssertTrue(try channel.finish().isClean)
     }
+
+    func testAddAfterForSynchronousPosition() {
+        let channel = EmbeddedChannel()
+        defer {
+            XCTAssertNoThrow(try channel.finish())
+        }
+
+        let firstHandler = IndexWritingHandler(1)
+        XCTAssertNoThrow(try channel.pipeline.addHandler(firstHandler).wait())
+        XCTAssertNoThrow(try channel.pipeline.addHandler(IndexWritingHandler(2)).wait())
+        XCTAssertNoThrow(
+            try channel.pipeline.addHandler(
+                IndexWritingHandler(3),
+                position: .after(firstHandler)
+            ).wait()
+        )
+
+        channel.assertReadIndexOrder([1, 3, 2])
+        channel.assertWriteIndexOrder([2, 3, 1])
+    }
+
+    func testAddBeforeForSynchronousPosition() {
```

Comment:
This test isn't actually using the synchronous API.

Reply:
🤦‍♂️